components/raftstore/src/store/snapshot_backup.rs

```diff
@@ -0,0 +1,345 @@
+    ) -> crate::coprocessor::Result<()> {
+        if self.allowed() {
+            return Ok(());
+        }
+        for req in reqs {
+            if req.has_ingest_sst() {
+                // Note: this will reject the batch of commands, which isn't so effective.
+                // But we cannot reject proposing a subset of command for now...
+                cx.bypass = true;
+                metrics::SNAP_BR_SUSPEND_COMMAND_TYPE
+                    .with_label_values(&["Ingest"])
+                    .inc();
+                return Err(self.reject());
+            }
+        }
+        Ok(())
+    }
+}
+
+impl AdminObserver for Arc<RejectIngestAndAdmin> {
+    fn pre_propose_admin(
+        &self,
+        _: &mut crate::coprocessor::ObserverContext<'_>,
+        admin: &mut kvproto::raft_cmdpb::AdminRequest,
+    ) -> crate::coprocessor::Result<()> {
+        if self.allowed() {
+            return Ok(());
+        }
+        if matches!(
+            admin.get_cmd_type(),
+            AdminCmdType::Split |
+            AdminCmdType::BatchSplit |
+            // We disable `CompactLog` here because if the log get truncated, 
+            // we may take a long time to send snapshots during restoring.
+            AdminCmdType::CompactLog |
+            // We will allow `Commit/RollbackMerge` here because the 
+            // `wait_pending_admin` will wait until the merge get finished.
+            // If we reject them, they won't be able to see the merge get finished.
+            // And will finally time out.
+            AdminCmdType::PrepareMerge |
+            AdminCmdType::ChangePeer |
+            AdminCmdType::ChangePeerV2
+        ) {
+            metrics::SNAP_BR_SUSPEND_COMMAND_TYPE
+                .with_label_values(&[&format!("{:?}", admin.get_cmd_type())])
+                .inc();
+            return Err(self.reject());
+        }
+        Ok(())
+    }
+
+    fn pre_transfer_leader(
+        &self,
+        _ctx: &mut crate::coprocessor::ObserverContext<'_>,
+        _tr: &kvproto::raft_cmdpb::TransferLeaderRequest,
+    ) -> crate::coprocessor::Result<()> {
+        if self.allowed() {
+            return Ok(());
+        }
+        metrics::SNAP_BR_SUSPEND_COMMAND_TYPE
+            .with_label_values(&["TransferLeader"])
+            .inc();
+        Err(self.reject())
+    }
+}
+
+/// A syncer for wait apply.
+/// The sender used for constructing this structure will:
+/// Be closed, if the `abort` has been called.
+/// Send the report id to the caller, if all replicas of this Syncer has been
+/// dropped.
+#[derive(Clone, Debug)]
+pub struct SnapshotBrWaitApplySyncer {
+    channel: Option<Arc<Mutex<Option<oneshot::Sender<u64>>>>>,
+    report_id: u64,
+}
+
+impl Drop for SnapshotBrWaitApplySyncer {
+    fn drop(&mut self) {
+        let chan = self
+            .channel
+            .take()
+            .expect("channel taken, maybe dropped twice");
+        if let Ok(ch) = Arc::try_unwrap(chan) {
```

Comment:
why `try_unwrap`

Reply:
This is a way to check whether this is the last one... I'm going to somehow refactor this code. You may review this part of code later.